### PR TITLE
Add horse health

### PR DESCRIPTION
**Pull request by MattRice12**
Added Horse Health feature where admins can schedule medical/care appointments for horses. The following additional features are included:
- Admins can write care notes
- Creating/editing/deleting appointments and care notes generates a notification
- Appointment Notifications are created according to the reminder intervals selected

### CODE REVIEW

**Comment by jah2488** [comment]
Hound is not happy. That's to be expected. 

However, it looks like the rspec tests are not passing on semaphore. Look into why that might be and we can work on the style issues after we are green.

File: app/models/horse_health.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          end
+        end
+      end
+      health_reminder_arr.push(["TODAY"]) if (health.due_date_days) == 0 && (health.due_date_weeks.blank?)
+      health_reminder_arr.each do |hr|
+        message = "APPOINTMENT REMINDER! #{health.horse.name} has an appointment #{hr[0]} (#{health.due.strftime('%D')}) with \"#{health.provider}\" for \"#{health.service}\"."
+        health.notify(health.user, message, 3, health)
+      end
+    end
+  end
```
Comment (jah2488) [inline app/models/horse_health.rb]: Is this the method that sends the reminders from the cron job?

File: spec/models/horse_health_spec.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      expect(note.notes).to eq("Hi there, I'm a note")
+    end
+
+    it "Reminder notification generated 1 day before due" do
+      current_user = User.first
+      health = HorseHealth.creation_notification(current_user, horse_health_hash) # Notification 1
+      health.due = Date.today + 1
+      health.save
+      HorseHealth.reminder_notifications # Notification 2
+      notification_message = "APPOINTMENT REMINDER! Hans has an appointment 1 day from today (#{health.due.strftime('%D')}) with \"FARRIER\" for \"Annual Shots\"."
```
Comment (jah2488) [inline spec/models/horse_health_spec.rb]: This `health.due.strftime('%D')` format for `due` should be its own method in horse health. Something like `formatted_due` or `due_str`.

**Comment by MattRice12** [comment]
That feel when rspec passes on my side but semaphore says it fails ....

**Comment by jah2488** [comment]
Where are we at with this? I see semaphore is failing again.

**Comment by MattRice12** [comment]
It failed Saturday evening. I took yesterday off but am about to start messing with it again.

**Comment by MattRice12** [comment]
Finally lol

**Comment by jah2488** [comment]
@MattRice12 Yayyyy.  Dates are always super tricky.

**Comment by jah2488** [comment]
All checks are passing, is this good to go now? (Meaning are you working on anything else right now?)

**Comment by MattRice12** [comment]
It is good to go. 👍

**Pull request merged**